### ISSUE TITLE
Brackets not working in C-BIOS

### ISSUE THREAD

**Issue by dgchrt** [openMSX/openMSX#1067]
I'm running openMSX on Mac (Sierra 10.12.5) and none of the brackets are working, either the normal square brackets nor the curly brackets by holding shift. Tried with different software.

**Comment by 0xe1f** [comment]
[FWIW](https://github.com/CocoaMSX/CocoaMSX/issues/44).

**Comment by MBilderbeek** [comment]
What kind of keyboard are you using on your Mac?

**Comment by dgchrt** [comment]
It's a U.S. MacBook, with standard U.S. key map on Settings.

**Comment by MBilderbeek** [comment]
Can you open the console (CMD+L) and type this command:
`set kbd_trace_key_presses on `
and then try to type these bracket keys and make a screenshot.

**Comment by mthuurne** [comment]
The link from pokebyte is very useful: I can reproduce the problem with both HotLogo and the Dutch MSX-Logo when using C-BIOS, but the problem doesn't occur when using the official MSX-BIOS.

The scan tables of C-BIOS do include brackets, so that is not the issue. But for some reason pressing the corresponding keys doesn't result in any characters appearing in Logo. The quote keys have the same issue, but other non-alphanumeric keys do work as expected.

**Comment by dgchrt** [comment]
Sure, @MBilderbeek:
![screen shot 2017-07-19 at 11 12 49 am](https://user-images.githubusercontent.com/1904261/28371435-6027ea06-6c73-11e7-8710-fc680576dae4.png)

@mthuurne : Brackets, semicolon, quotes and grave keys seem to have all the same issue.

**Comment by MBilderbeek** [comment]
Does the problem occur in MSX-BASIC with a normal BIOS?

After enabling that setting, press the nonfunctional keys and check the logging on stdout in de terminal in which you started openMSX.... and please paste that here. (Sorry, I misremembered that it was printed in the cnosole, but it isn't.)

**Comment by dgchrt** [comment]
Yes, the same issue happens on systems with their standard BIOS. Sony HB-F1XD is one of them. This is the output:
```
$ /Applications/openMSX.app/Contents/MacOS/openmsx ~/Documents/Misc/Software/MSX/Hot-Logo.rom 
Key pressed, unicode: 0x005b, keyCode: 0x0005b, keyName: LEFTBRACKET
Key released, unicode: 0x0000, keyCode: 0x40005b, keyName: LEFTBRACKET,RELEASE
Key pressed, unicode: 0x005d, keyCode: 0x0005d, keyName: RIGHTBRACKET
Key released, unicode: 0x0000, keyCode: 0x40005d, keyName: RIGHTBRACKET,RELEASE
```

**Comment by MBilderbeek** [comment]
Try it in MSX-BASIC of the HB-F1XD. Does it happen there too?

**Comment by MBilderbeek** [comment]
I can reproduce the issue in Hot Logo in combination with C-BIOS, as @mthuurne said. But I cannot reproduce the issue on the Sony HB-F1XD, with Hot Logo or in MSX-BASIC.... so that is strange.

Are you sure you were actually running that machine? In the command line you showed in your last post you are running the default machine (probably C-BIOS), unless you changed later in the console or OSD menu.

**Comment by dgchrt** [comment]
My bad, I was testing similar issues on another emulator and got some results mixed up. I have just tested Sony HB-F1XD running BASIC and the brackets are working fine. I can even run Hot-Logo in this machine without the brackets problem.

**Comment by MBilderbeek** [comment]
OK, so, the conclusion is that it is a bug in C-BIOS. Thanks for reporting!

**Comment by MBilderbeek** [comment]
Fixed: 6b8f117 Maarten ter Huurne: Update C-BIOS to v0.28

**Comment by 0xe1f** [comment]
@mthuurne Thanks for the note! :)
